### ISSUE TITLE
Incorrect Timezone

### ISSUE THREAD

**Issue by DmitriZhukov** [ClickHouse/ClickHouse#17638]
I am using DataGrip

use_server_time_zone | false
use_server_time_zone_for_dates | false
use_time_zone | Europe/Moscow

select timezone()
Europe/Berlin

Current Moscow time 17:45

select now()
2020-11-30 12:45:33

Why 5 hours difference?

**Comment by den-crane** [comment]
Maybe you have changed "VM options" ?

![image](https://user-images.githubusercontent.com/19737682/100629321-86875f00-32ff-11eb-8587-46b1306a0f67.png)

**Comment by DmitriZhukov** [comment]
No I dont have this setting

select datetime, toStartOfMinute(datetime) as minuteTime,

                                                     datetime >= '2020-11-30 00:00:00'  and
                                                     datetime <= '2020-11-30 15:38:00'  and
                                                 

2020-11-30 08:32:01,   2020-11-30 10:32:00

Why is that 2 hours difference?

Anyway I have a trade and its datetime is DG is **10:32:00**
when I run the same SQL from my application on server

return await _clickHouseClient.ExecuteQuerry<List<MinuteOperationsSum>>(ClickHouseQuery.GetOrganicTradesWithExchange(from, to, isMaker));

_select DISTINCT pair, toStartOfMinute(datetime) as minuteTime, sum(quantity) as qty, sum(cost) as paid, direction from organic_trades  where datetime >= '2020-11-30 00:00:00'  and datetime <= '2020-11-30 15:38:00'  and maker_trader in ('2b58cccc-5fbd-4154-9e73-a009d7c145c9','8fdd328f-47cc-4705-bf45-9a4f20314b87','ddee6a46-3bbd-4d39-b561-c028e618d8ff') group by pair, direction, toStartOfMinute(datetime) order by toStartOfMinute(datetime)_


it has time **11:32:00**

**Comment by den-crane** [comment]
please show the output `desc table organic_trades`

**Comment by DmitriZhukov** [comment]
date,Date,"","","","",""
datetime,DateTime,"","","","",""
timestamp,Int64,"","","","",""
id,String,"","","","",""
pair,String,"","","","",""
status,String,"","","","",""
direction,String,"","","","",""
maker_order,Nullable(String),"","","","",""
taker_order,Nullable(String),"","","","",""
maker_trader,Nullable(String),"","","","",""
taker_trader,Nullable(String),"","","","",""
price,Nullable(Float64),"","","","",""
quantity,Nullable(Float64),"","","","",""
cost,Nullable(Float64),"","","","",""
currency,String,"","","","",""
quote,String,"","","","",""
maker_fee,Nullable(Float64),"","","","",""
maker_fee_currency,String,"","","","",""
taker_fee,Nullable(Float64),"","","","",""
taker_fee_currency,String,"","","","",""

**Comment by den-crane** [comment]
Can you show `select version() ` ?

**Comment by DmitriZhukov** [comment]
20.11.2.1

**Comment by den-crane** [comment]
I don't see how it possible 

>select datetime, toStartOfMinute(datetime) as minuteTime,
>2020-11-30 08:32:01, 2020-11-30 10:32:00


for example
```sql
SELECT timezone()

┌─timezone()────┐
│ Europe/Berlin │
└───────────────┘

SELECT
    now() AS x,
    toStartOfMinute(x) AS y
┌───────────────────x─┬───────────────────y─┐
│ 2020-12-02 04:55:52 │ 2020-12-02 04:55:00 │
└─────────────────────┴─────────────────────┘

select now() x, toStartOfMinute(x) y format TSV
2020-12-02 04:57:47	2020-12-02 04:57:00
```

**Comment by alex-klepa** [comment]
I'm experiencing same issue:

from CH client, looks good:
```
SELECT version()
┌─version()─┐
│ 21.2.2.8  │
└───────────┘
 :) select timezone()

SELECT timezone()
┌─timezone()──────┐
│ America/Toronto │
└─────────────────┘

SELECT  now() AS x,  toStartOfMinute(x) AS y
┌───────────────────x─┬───────────────────y─┐
│ 2021-02-07 18:09:38 │ 2021-02-07 18:09:00 │
└─────────────────────┴─────────────────────┘
```

But from DataGrip or python client (https://github.com/mymarilyn/clickhouse-driver/):
```
>>> client.execute('select version() v, timezone() tz, now() x, toStartOfMinute(x) y;')
[array(['21.2.2.8', 'America/Toronto',
       datetime.datetime(2021, 2, 7, 18, 12, 4),
       datetime.datetime(2021, 2, 7, 23, 12)], dtype=object)]
```
![Screen Shot 2021-02-07 at 6 14 11 PM](https://user-images.githubusercontent.com/1385751/107162330-4fbd6d00-6970-11eb-8e13-51f2610b4ede.png)

**Comment by enqueue** [comment]
A reason for this might be that the driver shows a different result, when the server includes the time zone in the column information.

```
SELECT
    NOW(),
    toStartOfMinute(NOW())
FORMAT TabSeparatedWithNamesAndTypes
```
NOW() | toStartOfMinute(NOW()) 
-------| ---------------------
DateTime | DateTime(\'Europe/Berlin\') 
2021-02-15 08:45:17 | 2021-02-15 08:45:00

In the JDBC driver, the `use_server_time_zone` and `use_time_zone` settings only apply to "non-qualified" Date / DateTime fields. It also depends on how Datagrip uses the driver, e.g. if they call `getString`, which currently returns the wire-representation, or if they call `getDate` or `getObject`.

**Comment by zhicwu** [comment]
It's an issue in JDBC driver(see ClickHouse/clickhouse-jdbc#590) and it will be fixed in 0.3.0.
